resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v2/DriverPodKubernetesCredentialsMounter.scala

```diff
@@ -0,0 +1,173 @@
+   * that have to be created and mounted into the driver pod. The single Secret
+   * object contains all of the data entries for the driver pod's Kubernetes
+   * credentials. Returns empty if no secrets are to be mounted.
+   */
+  def createCredentialsSecret(): Option[Secret]
+
+  /**
+   * Mount any Kubernetes credentials from the submitting machine's disk into the
+   * driver pod.
+   */
+  def mountDriverKubernetesCredentials(
+    originalPodSpec: PodBuilder,
+    driverContainerName: String,
+    credentialsSecret: Option[Secret]): PodBuilder
+}
+
+private[spark] class DriverPodKubernetesCredentialsMounterImpl(
+    kubernetesAppId: String,
+    submitterLocalDriverPodKubernetesCredentials: KubernetesCredentials,
+    maybeUserSpecifiedMountedClientKeyFile: Option[String],
+    maybeUserSpecifiedMountedClientCertFile: Option[String],
+    maybeUserSpecifiedMountedOAuthTokenFile: Option[String],
+    maybeUserSpecifiedMountedCaCertFile: Option[String])
+    extends DriverPodKubernetesCredentialsMounter {
+
+  override def setDriverPodKubernetesCredentialLocations(sparkConf: SparkConf): SparkConf = {
+    val resolvedMountedClientKeyFile = resolveSecretLocation(
+      maybeUserSpecifiedMountedClientKeyFile,
+      submitterLocalDriverPodKubernetesCredentials.clientKeyDataBase64,
+      DRIVER_CREDENTIALS_CLIENT_KEY_PATH)
+    val resolvedMountedClientCertFile = resolveSecretLocation(
+      maybeUserSpecifiedMountedClientCertFile,
+      submitterLocalDriverPodKubernetesCredentials.clientCertDataBase64,
+      DRIVER_CREDENTIALS_CLIENT_CERT_PATH)
+    val resolvedMountedCaCertFile = resolveSecretLocation(
+      maybeUserSpecifiedMountedCaCertFile,
+      submitterLocalDriverPodKubernetesCredentials.caCertDataBase64,
+      DRIVER_CREDENTIALS_CA_CERT_PATH)
+    val resolvedMountedOAuthTokenFile = resolveSecretLocation(
+      maybeUserSpecifiedMountedOAuthTokenFile,
+      submitterLocalDriverPodKubernetesCredentials.oauthTokenBase64,
+      DRIVER_CREDENTIALS_OAUTH_TOKEN_PATH)
+    val sparkConfWithCredentialLocations = sparkConf.clone()
+      .setOption(KUBERNETES_DRIVER_MOUNTED_CA_CERT_FILE, resolvedMountedCaCertFile)
+      .setOption(KUBERNETES_DRIVER_MOUNTED_CLIENT_KEY_FILE, resolvedMountedClientKeyFile)
+      .setOption(KUBERNETES_DRIVER_MOUNTED_CLIENT_CERT_FILE, resolvedMountedClientCertFile)
+      .setOption(KUBERNETES_DRIVER_MOUNTED_OAUTH_TOKEN, resolvedMountedOAuthTokenFile)
+    sparkConfWithCredentialLocations.get(KUBERNETES_DRIVER_OAUTH_TOKEN).foreach { _ =>
+      sparkConfWithCredentialLocations.set(KUBERNETES_DRIVER_OAUTH_TOKEN, "<present_but_redacted>")
+    }
+    sparkConfWithCredentialLocations.get(KUBERNETES_SUBMIT_OAUTH_TOKEN).foreach { _ =>
+      sparkConfWithCredentialLocations.set(KUBERNETES_SUBMIT_OAUTH_TOKEN, "<present_but_redacted>")
+    }
+    sparkConfWithCredentialLocations
+  }
+
+  override def createCredentialsSecret(): Option[Secret] = {
+    val allSecretData = resolveSecretData(
```

Comment:
nit: drop `resolveSecretData` onto the next line so it lines up with the other `resolveSecretData`s